### PR TITLE
[Refactor] refactor ddl executor createFunction & dropFunction

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [x] refactor

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

refactor ddl executor createFunction & dropFunction to make GlobalStateMgr simple

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 26 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/execution/CreateFunctionExecutor.java|0|12|00.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/execution/DropFunctionExecutor.java|0|12|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/execution/DropFunctionExecutor.java [hunk 1]
```diff
@@ -3,15 +3,36 @@
...
+        FunctionName name = dropFunctionStmt.getFunctionName();
+        Database db = context.getGlobalStateMgr().getDb(name.getDb());
+        if (db == null) {
+            ErrorReport.reportDdlException(ErrorCode.ERR_BAD_DB_ERROR, name.getDb());
+        }
+        db.dropFunction(dropFunctionStmt.getFunction());
         return null;
     }
+
+    public static void replayDropFunction(FunctionSearchDesc functionSearchDesc) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/execution/DropFunctionExecutor.java:30]: This replay logic should be put in Database.java

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/execution/DropFunctionExecutor.java:30]: ok，I will move it

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 22 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/execution/CreateFunctionExecutor.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/execution/DropFunctionExecutor.java|0|6|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java [hunk 2]
```diff
@@ -637,12 +637,22 @@ public static void loadJournal(GlobalStateMgr globalStateMgr, JournalEntity jour
                 }
                 case OperationType.OP_ADD_FUNCTION: {
                     final Function function = (Function) journal.getData();
-                    GlobalStateMgr.getCurrentState().replayCreateFunction(function);
+                    String dbName = function.getFunctionName().getDb();
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java:640]: How about add static Function replayCreateFunctionLog() and replayDropFunction(). and make these logic in the function
And call Database.replayCreateFunctionLog()?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 26 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|12|00.00%|
|:large_blue_circle:|com/starrocks/execution/CreateFunctionExecutor.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/execution/DropFunctionExecutor.java|0|6|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java (hunk 2 above)
Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/persist/EditLog.java]: I did it

**Review by imay — APPROVED** [review]

**Pull request merged**